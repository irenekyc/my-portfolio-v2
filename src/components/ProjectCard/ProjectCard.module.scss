.ikProjectCard {
  &__container {
    width: 320px;
    margin: 12px;
    background: #eee;
    color: #303841;
    height: 420px;
    padding: 1.5rem 1rem;
    border-radius: 25px;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.5);
    box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.5);
  }
  &__title {
    &__div {
      height: 3.5rem;
      margin-bottom: 0.25rem;
      display: flex;
      align-items: flex-end;
    }
    font-size: 1.5rem;
    line-height: 1;
  }
  &__heading {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    margin-bottom: 0.25rem;

    &__tags {
      display: flex;
      > * {
        margin-right: 0.5rem;
      }
      i {
        font-size: 40px;
      }
    }
    &__links {
      a {
        margin-right: 0.5rem;
        color: #303841;
        font-size: 1rem;

        &:last-child {
          margin-right: 0;
        }
      }
      display: flex;
    }
  }
  &__image {
    margin-bottom: 0.25rem;
    height: 150px;
    overflow: hidden;
  }
  &__excerpt {
    display: -webkit-box;
    max-width: 300px;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    overflow: hidden;
    margin-bottom: 0.25rem;
  }

  &__learnMore {
    font-size: 14px;
    padding: 0.1rem 0.5rem;
  }
}
