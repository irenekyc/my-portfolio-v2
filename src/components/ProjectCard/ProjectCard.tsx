import React, { useState, useEffect, FunctionComponent } from "react";
import MainTagIcon from "../MainTagIcon/MainTagIcon";
import { CSSTransition } from "react-transition-group";
import { Project } from "../../typings/Project";

interface ProjectCardProps {
  data: Project;
  onClickOpenDetails: (projectDetails: Project) => void;
}

const ProjectCard: FunctionComponent<ProjectCardProps> = ({
  data,
  onClickOpenDetails,
}: ProjectCardProps) => {
  const [ready, setReady] = useState(false);

  useEffect(() => {
    if (data) {
      setReady(true);
    } else if (data == null) {
      setReady(false);
    }
  }, [data]);

  return (
    <CSSTransition in={ready} timeout={500} unmountOnExit classNames="fadeIn">
      <div className="portfolio-card">
        <div className="portfolio-card-heading">
          <div className="portfolio-card-heading-main">
            {data.mainTag.map((tag) => (
              <MainTagIcon tag={tag} key={`${data.title}__${tag}`} />
            ))}
          </div>
          <div className="portfolio-card-heading-links">
            <a href={data.projectUrl} rel="noopener noreferrer" target="_blank">
              <i className="fas fa-globe"></i>
            </a>
            <a href={data.github} rel="noopener noreferrer" target="_blank">
              <i className="fab fa-github"></i>
            </a>
          </div>
        </div>
        <div className="portfolio-card-title">
          <h1>{data.title}</h1>
        </div>
        <div className="portfolio-card-image">
          <img src={`images/${data.image}`} alt="" />
        </div>
        <div className="portoflio-card-content">
          <p>{data.excerpt}</p>
        </div>
        <div className="portfolio-card-learn-more">
          <button onClick={() => onClickOpenDetails(data)} id={data.title}>
            {" "}
            Learn more
          </button>
        </div>
      </div>
    </CSSTransition>
  );
};

export default ProjectCard;
