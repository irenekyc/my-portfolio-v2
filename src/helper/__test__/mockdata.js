export const mockProject = {
  title: "title",
  category: "category",
  tools: [],
  mainTag: [],
  image: "image.png",
  excerpt: "some excerpt",
  fullDescription: "some description",
  projectUrl: undefined,
  github: "https://github.com",
};

export const mockProjectListLong = [
  {
    title: "title1",
    category: "category",
    tools: ["Next"],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title2",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title3",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title4",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title5",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title6",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title7",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title8",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },

  {
    title: "title9",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title10",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
  {
    title: "title11",
    category: "category",
    tools: [],
    mainTag: [],
    image: "image.png",
    excerpt: "some excerpt",
    fullDescription: "some description",
    projectUrl: undefined,
    github: "https://github.com",
  },
];
