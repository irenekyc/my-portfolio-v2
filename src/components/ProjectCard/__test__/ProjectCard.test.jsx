import React from "react";
import ProjectCard from "../index";
import { render, screen, fireEvent } from "@testing-library/react";
import { mockProject } from "../../../helper/__test__/mockdata";

const mockOpenModal = jest.fn();

it("Render ProjectCard", () => {
  render(<ProjectCard data={mockProject} onClickOpenDetails={mockOpenModal} />);

  const projectCard = screen.getByTestId("portfolio-project-card");
  expect(projectCard).toBeInTheDocument();

  const projectTitle = screen.getByTestId("project-card-title");
  expect(projectTitle).toHaveTextContent(mockProject.title);
});

it("Render ProjectCard - click on learn more button", () => {
  render(<ProjectCard data={mockProject} onClickOpenDetails={mockOpenModal} />);

  const projectCard = screen.getByTestId("portfolio-project-card");
  expect(projectCard).toBeInTheDocument();

  const learnMoreButton = screen.getByTestId("project-card-learn-more");
  expect(learnMoreButton).toBeInTheDocument();
  fireEvent.click(learnMoreButton);
  expect(mockOpenModal).toBeCalledWith(mockProject);
});
