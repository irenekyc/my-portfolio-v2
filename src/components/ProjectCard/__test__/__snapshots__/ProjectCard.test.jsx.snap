// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ProjectCard - click on learn more button 1`] = `
<div
  class="ikProjectCard__container fadeIn-enter fadeIn-enter-active"
  data-testid="portfolio-project-card"
>
  <div
    class="ikProjectCard__heading"
  >
    <div
      class="ikProjectCard__heading__tags"
    />
    <div
      class="ikProjectCard__heading__links"
    >
      <a
        data-testid="project-card-github"
        href="https://github.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          class="fab fa-github"
        />
      </a>
    </div>
  </div>
  <div
    class="ikProjectCard__title__div"
  >
    <h2
      class="ikProjectCard__title"
      data-testid="project-card-title"
    >
      title
    </h2>
  </div>
  <div
    class="ikProjectCard__image"
  >
    <img
      alt=""
      data-testid="project-card-image"
      src="images/image.png"
    />
  </div>
  <div
    class="ikProjectCard__excerpt"
    data-testid="project-card-excerpt"
  >
    <p>
      some excerpt
    </p>
  </div>
  <div>
    <button
      class="ikProjectCard__learnMore"
      data-testid="project-card-learn-more"
      id="title"
    >
       
      Learn more
    </button>
  </div>
</div>
`;
